Tests/LibCore/TestLibCoreIODevice.cpp

```diff
@@ -76,3 +76,23 @@ TEST_CASE(file_get_read_position)
         EXPECT_EQ(offset, 0);
     }
 }
+
+TEST_CASE(file_lines_range)
+{
+    auto path = "long_lines.txt";
+    auto file_or_error = Core::File::open(path, Core::OpenMode::ReadOnly);
+    if (file_or_error.is_error()) {
+        warnln("Failed to open {}: {}", path, file_or_error.error());
+        VERIFY_NOT_REACHED();
+    }
+    auto file = file_or_error.release_value();
+    auto output_path = "/tmp/output.txt";
+    auto outfile_or_error = Core::File::open(output_path, Core::OpenMode::WriteOnly);
+    auto outputfile = outfile_or_error.release_value();
+    for (auto line : file->lines()) {
+        outputfile->write(line);
+        outputfile->write("\n");
```

Comment:
Does the iterator strip out newlines the instead of just splitting lines on them? That seems like kind of strange behavior?

Reply:
that behaviour is from `LineIterator`, and I can honestly think of use-cases for both.
not sure how either would be strange tbh

Reply:
Sure, fair enough.